src/helm/proxy/clients/http_model_client.py

```diff
@@ -118,23 +118,23 @@ def do_it():
         )
 
     def decode(self, request: DecodeRequest) -> DecodeRequestResult:
-        raise NotImplementedError("Not implemented yet.")
-        # cache_key = asdict(request)
-
-        # try:
-
-        #     def do_it():
-        #         url = f"{self.base_url}/decode"
-        #         response = requests.post(url, json={"tokens": request.tokens})
-        #         response.raise_for_status()
-        #         response_data = response.json()
-        #         return response_data
-
-        #     result, cached = self.cache.get(cache_key, wrap_request_time(do_it))
-        # except Exception as e:
-        #     error: str = f"Local Model error: {e}"
-        #     return DecodeRequestResult(success=False, cached=False, error=error, text="")
-
-        # return DecodeRequestResult(
-        #     success=True, cached=cached, text=result["text"], request_time=result["request_time"]
-        # )
+        # raise NotImplementedError("Not implemented yet.")
+        cache_key = asdict(request)
+
+        try:
+
+            def do_it():
+                url = f"{self.base_url}/decode"
+                response = requests.post(url, json={"tokens": request.tokens})
+                response.raise_for_status()
+                response_data = response.json()
+                return response_data
+
+            result, cached = self.cache.get(cache_key, wrap_request_time(do_it))
```

Comment:
This should be:
```

            if self.cache:
                result, cached = self.cache.get(cache_key, wrap_request_time(do_it))
            else:
                result, cached = do_it(), False
```
which is similar to `make_reqeuest()` / `tokenize()`

Reply:
you are right, sorry this was just uncommenting. I am still trying to find a run-spec that exercises this decode request